include/sleipnir/control/OCPSolver.hpp

```diff
@@ -0,0 +1,76 @@
+
+
+#pragma once
+
+#include <Eigen/Core>
+
+#include "sleipnir/SymbolExports.hpp"
+#include "sleipnir/optimization/OptimizationProblem.hpp"
+#include "sleipnir/autodiff/VariableMatrix.hpp"
+
+namespace sleipnir {
+
+typedef std::function<VariableMatrix(double, VariableMatrix, VariableMatrix)> FixedStepODE;
+typedef std::function<VariableMatrix(VariableMatrix, VariableMatrix)> FixedStepObjective;
+typedef std::function<void(VariableMatrix, VariableMatrix, bool)> FixedStepConstraintFunction;
+
+/**
+ * Performs 4th order Runge-Kutta integration of dx/dt = f(t, x, u) for dt.
+ *
+ * @param f  The function to integrate. It must take two arguments x and u.
+ * @param x  The initial value of x.
+ * @param u  The value u held constant over the integration period.
+ * @param t0 The initial time.
+ * @param dt The time over which to integrate.
+ */
+template <typename T, typename U, typename V>
+T RK4(std::function<T(V,T,U)> f, T x, U u, V t0, V dt) {
+  auto halfdt = dt * 0.5;
+  T k1 = f(t0, x, u);
+  T k2 = f(t0 + halfdt, x + halfdt * k1, u);
+  T k3 = f(t0 + halfdt, x + halfdt * k2, u);
+  T k4 = f(t0 + dt, x + dt * k3, u);
+
+  return x + (k1 + k2 * 2. + k3 * 2. + k4) * (dt / 6.0);
+}
+
+class SLEIPNIR_DLLEXPORT FixedStepOCPSolver {
+  public:
+  FixedStepOCPSolver(int N, int M, double dt, int numSteps) noexcept;
+  void SetExplicitF(FixedStepODE f) { m_F = f; };
+  void SetLinearDiscrete(Eigen::MatrixXd Ad, Eigen::MatrixXd Bd) { m_Ad = Ad; m_Bd = Bd; };
+  void ConstrainInitialState(const VariableMatrix initialState) { m_problem.SubjectTo(X().Col(0) == initialState); };
+  void ConstrainFinalState(const VariableMatrix finalState) { m_problem.SubjectTo(X().Col(m_numSteps) == finalState); };
```

Comment:
Did you mean to use `const VariableMatrix&` here instead? Putting const on copies does nothing.